benchmark/pkg/comparer/schemes/avgtest.go

```diff
@@ -0,0 +1,48 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package schemes
+
+import (
+	"fmt"
+	"math"
+
+	"k8s.io/perf-tests/benchmark/pkg/util"
+)
+
+// CompareJobsUsingAvgTest takes a JobComparisonData object, compares left
+// and right jobs for each metric inside it and fills in the comparison
+// results in the metric's object after checking ratio of the averages
+// of its left and right samples is within the allowed ratio lower bound
+// and upper bound (which is the inverse of lower bound).
+func CompareJobsUsingAvgTest(jobComparisonData *util.JobComparisonData, allowedRatioLowerBound float64) {
+	jobComparisonData.ComputeStatsForMetricSamples()
+	for _, metricData := range jobComparisonData.Data {
+		leftSampleCount := len(metricData.LeftJobSample)
+		rightSampleCount := len(metricData.RightJobSample)
+		metricData.Matched = false
+		var leftRightAvgRatio float64
+		if leftSampleCount == 0 || rightSampleCount == 0 {
+			leftRightAvgRatio = math.NaN()
+		} else {
+			leftRightAvgRatio = metricData.AvgL / metricData.AvgR
+			if allowedRatioLowerBound <= leftRightAvgRatio && leftRightAvgRatio <= 1/allowedRatioLowerBound {
+				metricData.Matched = true
+			}
+		}
+		metricData.Comments += fmt.Sprintf("Avg_ratio=%.4f\t\tN_left=%v\tN_right=%v", leftRightAvgRatio, leftSampleCount, rightSampleCount)
```

Comment:
There seems to be no good reason of using tabwriter here. It makes sense when we want to print multiple rows of text with columns that are tab-aligned. Here we just are generating a single line of comment (which doesn't know how the other comments are going to look like in order to align). Am I missing something here?